pinot-tools/src/main/java/org/apache/pinot/tools/tuner/strategy/OLSAnalysisImpl.java

```diff
@@ -0,0 +1,348 @@
+        String.format("%.0f", timePercentile[2]), String.format("%.0f", timePercentile[3]), String.format("%.0f", timePercentile[4]), String.format("%.0f", timePercentile[5]),
+        String.format("%.0f", timePercentile[6]), String.format("%.0f", timePercentile[7]), String.format("%.0f", timePercentile[8]), String.format("%.0f", timePercentile[9]));
+
+    OLSMultipleLinearRegression regression = new OLSMultipleLinearRegression();
+    regression.setNoIntercept(true);
+
+    double[] time = new double[olsMergerObj.getMinBin().size()];
+    double[][] x_arr = new double[olsMergerObj.getMinBin().size()][2];
+    AtomicInteger iter = new AtomicInteger(0);
+
+    olsMergerObj.getMinBin().forEach((key, val) -> {
+      time[iter.get()] = val._2();
+      x_arr[iter.get()][0] = key._1() * _lenBin + _lenBin / 2;
+      x_arr[iter.get()][1] = key._2() * _lenBin + _lenBin / 2;
+      iter.incrementAndGet();
+    });
+
+    try {
+      regression.newSampleData(time, x_arr);
+      double[] params = regression.estimateRegressionParameters();
+      double rSquared = regression.calculateRSquared();
+      if (rSquared > MODEL_R_SQUARE_THRESHOLD) {
+        reportOut += "\nMaximum Optimization(ms):\n";
+        reportOut += MessageFormat.format(
+            "10%:{0} | 20%:{1} | 30%:{2} | 40%:{3} | 50%:{4} | 60%:{5} | 70%:{6} | 80%:{7} | 90%:{8} | 95%:{9}\n",
+            String.format("%.0f", numEntriesScannedInFilterPercentile[0] * params[0]),
+            String.format("%.0f", numEntriesScannedInFilterPercentile[1] * params[0]),
+            String.format("%.0f", numEntriesScannedInFilterPercentile[2] * params[0]),
+            String.format("%.0f", numEntriesScannedInFilterPercentile[3] * params[0]), String.format("%.0f", numEntriesScannedInFilterPercentile[4] * params[0]),
+            String.format("%.0f", numEntriesScannedInFilterPercentile[5] * params[0]), String.format("%.0f", numEntriesScannedInFilterPercentile[6] * params[0]),
+            String.format("%.0f", numEntriesScannedInFilterPercentile[7] * params[0]), String.format("%.0f", numEntriesScannedInFilterPercentile[8] * params[0]),
+            String.format("%.0f", numEntriesScannedInFilterPercentile[9] * params[0]));
+        reportOut += MessageFormat.format("\nR-square: {0}\n", rSquared);
+        reportOut += String.format("Params: %s %s\n", Double.toString(params[0]), Double.toString(params[1]));
+      } else {
+        reportOut += "\nunable to predict the optimization boundary of this table!";
+      }
+    } catch (MathIllegalArgumentException | MathArithmeticException e) {
+      reportOut += "\nunable to predict the optimization boundary of this table!";
+    }
+    LOGGER.info(reportOut);
+  }
+
+  /**
+   * Parse and count the total bitmaps used in a query
+   */
+  class DimensionScoring {
```

Comment:
why is DimensionScoring here? I don't see it being used in this class anywhere

Reply:
Not for now but will be useful for future study of overhead of or-ing a very long list of inverted indices

Reply:
Not for now, but could be very useful to study the overhead of or-ing a very long list of inverted indices